src/lib/ext-activitylog.js

```diff
@@ -65,9 +112,28 @@ class Controller {
     const existingLogs = await this.getExistingLogs();
 
     if (existingLogs.length) {
-      this.model.addNewLogs(existingLogs);
-      this.view.addTableRows(this.model.logs);
+      this.handleNewLogs(existingLogs);
+    }
+  }
+
+  handleNewLogs(logs) {
+    for (const log of logs) {
+      this.model.addNewLog(log);
+      this.isFilterMatched(this.model.filter, log)
+        ? this.view.handleNewLog({ log, isHidden: true })
+        : this.view.handleNewLog({ log, isHidden: false });
+    }
+  }
+
+  isFilterMatched(filter, log) {
+    for (const key of Object.keys(filter)) {
+      if (Object.keys(log).includes(key)) {
```

Comment:
> ... in this specific case, I would actually get rid of the check altogether. Every key should be present, and if not you can filter by `undefined` instead.

I am not sure if I understand the last line of the comment.
 Since I am returning boolean, I deal with the `undefined` filter property [here](https://github.com/mozilla/extension-activity-monitor/pull/10/files#diff-9f25b3c1ffafaec3c4d02ac711c6392aR69) to keep it more natural. Is this okay ?

Reply:
> >Every key should be present, and if not you can filter by `undefined` instead.
>
> I am not sure if I understand the last line of the comment.

`Object.keys(log).includes(key)` checks if the key is present in the log. The current keys to filter on are present in the log, so the check is redundant. But even if the key was missing, and you want a way to filter those entries, then you can filter by `undefined` instead (like you're doing in the code snippet that you had linked).